packages/utils/src/chromium-finder.ts

```diff
@@ -0,0 +1,391 @@
+             */
+            let execPaths;
+
+            /*
+             * Some systems do not support grep -R so fallback to -r.
+             * See https://github.com/GoogleChrome/chrome-launcher/issues/46 for more context.
+             */
+            try {
+                execPaths = execSync(`grep -ER "${chromeExecRegex}" ${folder} | awk -F '=' '{print $2}'`);
+            } catch (e) {
+                execPaths = execSync(`grep -Er "${chromeExecRegex}" ${folder} | awk -F '=' '{print $2}'`);
+            }
+
+            execPaths = execPaths.toString()
+                .split(newLineRegex)
+                .map((execPath: string) => {
+                    return execPath.replace(argumentsRegex, '$1');
+                });
+
+            for (const execPath of execPaths) {
+                if (canAccess(execPath)) {
+                    return execPath;
+                }
+            }
+        }
+    }
+
+    return '';
+};
+
+/**
+ * Look for linux executables in 2 ways
+ * 1. Look into the directories where .desktop are saved on gnome based distro's
+ * 2. Look for google-chrome-stable & google-chrome executables by using the which command
+ */
+const linux = (browser: Browser) => {
+
+    // 2. Look into the directories where .desktop are saved on gnome based distro's
+    const desktopInstallationFolders = [
+        path.join(require('os').homedir(), '.local/share/applications/'),
+        '/usr/share/applications/'
+    ];
+
+    const browserPartsInfo = browserVariables.get('linux');
+    const executables = browserPartsInfo!.get(browser);
+
+    if (!executables) {
+        throw new Error(ERRORS.NotSupportedBrowser);
+    }
+
+    for (const folder of desktopInstallationFolders) {
+        const executable = findChromeExecutable(folder, executables);
+
+        if (executable) {
+            return executable;
+        }
+    }
+
+    // Look for executables by using the which command
+    for (const executable of executables) {
+        try {
+            const chromePath =
+                execFileSync('which', [executable], { stdio: 'pipe' }).toString()
+                    .split(newLineRegex)[0];
+
+            if (chromePath && canAccess(chromePath)) {
+                return chromePath;
+            }
+        } catch (e) {
+            // Not installed.
+        }
+    }
+
+    return '';
+};
+
+/**
+ * Searches for installation paths for the given `browser` on a Windows platform.
+ * @param browser The type of browser to search for
+ */
+const win32 = (browser: Browser) => {
+    const info = browserVariables.get('win32');
+
+    if (!info) {
+        throw new Error(ERRORS.NotSupportedBrowser);
+    }
+
+    const suffixes = info.get(browser);
+
+    if (!suffixes) {
+        throw new Error(ERRORS.NotSupportedBrowser);
+    }
+
+    const prefixes = [
+        getVariable('LOCALAPPDATA'),
+        getVariable('PROGRAMFILES'),
+        getVariable('PROGRAMFILES(X86)')
+    ].filter(Boolean);
```

Comment:
A lot of this file is [copy pasta from chrome-finder in the chrome-launcher project](https://github.com/GoogleChrome/chrome-launcher/blob/master/src/chrome-finder.ts). In this specific case it's the `Boolean` constructor which will filter any environment variable that is not defined or returns `''`.

It also took me a moment to figure it out when I was looking the other day but because we might contribute changes upstream I didn't want to change too much their logic.

Reply:
Ahh, ok, it is the first time I see that xD